q2_longitudinal/_longitudinal.py

```diff
@@ -276,18 +273,20 @@ def _volatility(output_dir, metadata, state_column, individual_id_column,
         raise ValueError('state_column must contain at least two unique '
                          'values.')
 
-    control_chart_data = metadata.to_dataframe()
-    # convert np.nan to None (nans and vega don't mix)
-    control_chart_data = _convert_nan_to_none(control_chart_data)
-    # If we made it this far that means we can let Vega do it's thing!
     group_columns = list(categorical.columns.keys())
     if individual_id_column and individual_id_column not in group_columns:
         group_columns += [individual_id_column]
+        if individual_id_column not in metadata.columns.keys():
```

Comment:
individual_id_column is added back to metadata if it is missing (e.g., because it is numeric and gets dropped)